doctr/models/kie_predictor/pytorch.py

```diff
@@ -71,27 +71,34 @@ def forward(
 
         origin_page_shapes = [page.shape[:2] if isinstance(page, np.ndarray) else page.shape[-2:] for page in pages]
 
+        # Localize text elements
+        loc_preds, out_maps = self.det_predictor(pages, return_maps=True, **kwargs)
+
         # Detect document rotation and rotate pages
+        seq_maps = [
+            np.sum(np.where(out_map > kwargs.get("bin_thresh", 0.3), 255, 0), axis=-1, keepdims=True).astype(np.uint8)
+            for out_map in out_maps
+        ]
         if self.detect_orientation:
-            origin_page_orientations = [estimate_orientation(page) for page in pages]  # type: ignore[arg-type]
+            origin_page_orientations = [estimate_orientation(seq_map) for seq_map in seq_maps]
             orientations = [
-                {"value": orientation_page, "confidence": 1.0} for orientation_page in origin_page_orientations
+                {"value": orientation_page, "confidence": None} for orientation_page in origin_page_orientations
             ]
         else:
             orientations = None
         if self.straighten_pages:
             origin_page_orientations = (
                 origin_page_orientations
                 if self.detect_orientation
-                else [estimate_orientation(page) for page in pages]  # type: ignore[arg-type]
+                else [estimate_orientation(seq_map) for seq_map in seq_maps]
             )
             pages = [
                 rotate_image(page, -angle, expand=True)  # type: ignore[arg-type]
                 for page, angle in zip(pages, origin_page_orientations)
             ]
+            # forward again to get predictions on straight pages
```

Comment:
Rotate instead of forward again https://github.com/mindee/doctr/blob/8e54989845f4256fdae7ac4f3794a4cd53682b1e/doctr/utils/geometry.py#L102